### ISSUE TITLE
Indentation error: List in a map in a list

### ISSUE THREAD

**Issue by vrivellino** [adrienverge/yamllint#97]
I'm not sure if this a duplicate of #80, but I'm seeing what looks like a spurious error.

Test file:
```yaml
---

Level1:
  Level2:
    Type: test object
    Properties:
      prop1: one
      prop2: two
      prop3:
        list1:
          - itemA
          - [sublist1_itemA, sublist1_itemB]
          - list2:
            - itemB
            - [sublist2_itemA, sublist2_itemB]
            - [sublist3_itemA, sublist3_itemB]
```

yamllint output:
```
$ yamllint test.
yml 
test.yml
  14:13     error    wrong indentation: expected 14 but found 12  (indentation)
```

That error is due to the indentation of `itemB`.

Is that not valid?

If one were to convert it json, this is the object I'd need to produce:

```json
{
  "Level1": {
    "Level2": {
      "Type": "test object",
      "Properties": {
        "prop1": "one",
        "prop2": "two",
        "prop3": {
          "list1": [
            "itemA",
            [ "sublist1_itemA", "sublist1_itemB" ],
            { "list2": [
              "itemB",
              [ "sublist2_itemA", "sublist2_itemB" ],
              [ "sublist3_itemA", "sublist3_itemB" ]
            ]}
          ]
        }
      }
    }
  }
}
```

**Comment by adrienverge** [comment]
Hi,

I think it's the same as in #80.

You should use either this:
```yaml
list1:
  - itemA
  - [sublist1_itemA, sublist1_itemB]
  - list2:
      - itemB
      - [sublist2_itemA, sublist2_itemB]
      - [sublist3_itemA, sublist3_itemB]
```

or that:
```yaml
list1:
- itemA
- [sublist1_itemA, sublist1_itemB]
- list2:
  - itemB
  - [sublist2_itemA, sublist2_itemB]
  - [sublist3_itemA, sublist3_itemB]
```

Using an intermediate indentation style (like in your example) works in most cases, but can lead to non-sense invalid syntaxes (see [this comment](https://github.com/adrienverge/yamllint/issues/80#issuecomment-339718327)), hence is not supported by yamllint.

I hope this helps!

**Comment by vrivellino** [comment]
> Using an intermediate indentation style

Sorry, I'm a bit confused. What do you mean by that?

Using your first example, shifting the indentation of itemB (and sublist2 & sublist3) does quiet the error, although I don't understand why.

However, unshifting the indentation of the list items (as in your second example) produces the indentation warning at the lines for both `itemA` and `itemB`:

```yaml
---

Level1:
  Level2:
    Type: test object
    Properties:
      prop1: one
      prop2: two
      prop3:
        list1:
        - itemA
        - [sublist1_itemA, sublist1_itemB]
        - list2:
          - itemB
          - [sublist2_itemA, sublist2_itemB]
          - [sublist3_itemA, sublist3_itemB]
```

```
$ yamllint test.yml
test.yml
  11:9      error    wrong indentation: expected 10 but found 8  (indentation)
  14:11     error    wrong indentation: expected 12 but found 10  (indentation)
```

Thanks for the response, much appreciated!

**Comment by adrienverge** [comment]
Hi, sorry if I was unclear!

*"Using an intermediate indentation style"* would be, for instance, indenting items for `list1` but not those of `list2` (exactly like in your example of the original post).

I think what you want is to set `indent-sequences: false` in yamllint configuration. Have you tried [creating your own config file](https://yamllint.readthedocs.io/en/stable/configuration.html#extending-the-default-configuration) and [customizing the `indentation` rule](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.indentation)?

In short, create a `.yamllint` file in your current directory and paste this:
```yaml
rules:
  indentation:
    spaces: 2
    indent-sequences: false
    check-multi-line-strings: false
```
Then, the file `test.yaml` of your latest example will pass without problem.

If you really want to mix indented lists and non-indented lists (like in your original post example), use `indent-sequences: whatever`.
